controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -1122,6 +1124,177 @@ bool master_thread::handle_cmdu_1905_channel_selection_response(Socket *sd,
+                       << std::endl;
+
+            //fill tx data from TLV
+            if (!new_link_metric_data.add_transmitter_link_metric(TxLinkMetricData)) {
+                LOG(ERROR) << "adding TxLinkMetricData has failed";
+                return false;
+            }
+        }
+
+        if (tlvType == int(ieee1905_1::eTlvType::TLV_RECEIVER_LINK_METRIC)) {
+            // parse tlvReceiverLinkMetric
+            auto RxLinkMetricData = cmdu_rx.addClass<ieee1905_1::tlvReceiverLinkMetric>();
+            if (!RxLinkMetricData) {
+                LOG(ERROR) << "addClass ieee1905_1::tlvReceiverLinkMetric has failed";
+                return false;
+            }
+
+            if (!network_utils::mac_compare(
+                    reporting_agent_al_mac,
+                    RxLinkMetricData->al_mac_of_the_device_that_transmits())) {
+                LOG(ERROR) << "TLV_RECEIVER_LINK_METRIC reporter al_mac ="
+                           << network_utils::mac_to_string(reporting_agent_al_mac) << std::endl
+                           << " and TLV_TRANSMITTER_LINK_METRIC reporter al_mac ="
+                           << network_utils::mac_to_string(
+                                  RxLinkMetricData->al_mac_of_the_device_that_transmits())
+                           << std::endl
+                           << " not the same";
+                return false;
+            }
+
+            if (!network_utils::mac_compare(
+                    neighbor_al_mac,
+                    RxLinkMetricData
+                        ->al_mac_of_the_neighbor_whose_link_metric_is_reported_in_this_tlv())) {
+                LOG(ERROR)
+                    << "TLV_RECEIVER_LINK_METRIC reported al_mac ="
+                    << network_utils::mac_to_string(neighbor_al_mac) << std::endl
+                    << " and TLV_TRANSMITTER_LINK_METRIC reported al_mac ="
+                    << network_utils::mac_to_string(
+                           RxLinkMetricData
+                               ->al_mac_of_the_neighbor_whose_link_metric_is_reported_in_this_tlv())
+                    << std::endl
+                    << " not the same";
+                return false;
+            }
+
+            LOG(DEBUG) << "recieved  tlvReceiverLinkMetric from al_mac="
+                       << network_utils::mac_to_string(reporting_agent_al_mac) << std::endl
+                       << "reported  al_mac =" << network_utils::mac_to_string(neighbor_al_mac)
+                       << std::endl;
+
+            //fill rx data from TLV
+            if (!new_link_metric_data.add_receiver_link_metric(RxLinkMetricData)) {
+                LOG(ERROR) << "adding RxLinkMetricData has failed";
+                return false;
+            }
+        }
+    }
+
+    //save agent link metric data to database or modify existing.
+    if (m_link_metric_data.find(reporting_agent_al_mac) != m_link_metric_data.end()) {
+        if (m_link_metric_data[reporting_agent_al_mac].find(neighbor_al_mac) !=
```

Comment:
`find()` method already returns an iterator to the element, so it is best to use what you find instead of look it up again with `[]` operator.

Reply:
second find is for the second map (its a map[reporter] of a map[neighbor])

Reply:
You didn't understand me. I'll try to explain with code:
```cpp
auto neigbour_map_it = m_link_metric_data.find(reporting_agent_al_mac;
if (neigbour_map_it = m_link_metric_data.find(reporting_agent_al_mac) != m_link_metric_data.end()) {
    if (neigbour_map_it->second.find(something))
```